.github/workflows/test.yml

```diff
@@ -23,6 +23,27 @@ jobs:
           cache: npm
           node-version-file: ".nvmrc"
 
+      - name: Checkout base package-lock.json
+        uses: actions/checkout@v5
+        with:
+          path: base
+          ref: ${{ github.base_ref }}
+          sparse-checkout: |
+            package-lock.json
+          sparse-checkout-cone-mode: false
+
+      - name: Check package-lock.json is correct
+        run: |
+          if ! diff -q package-lock.json base/package-lock.json; then
+            mv package-lock.json package-lock.json.branch
+            mv base/package-lock.json .
+            rm -rf node_modules
+            npm install
+            diff package-lock.json package-lock.json.branch
```

Comment:
If I understand this correctly, then this assumes that running `npm install` results in the same `package-lock.json`. However, if a PR adds a dependency (with a subtree of transitive dependencies), then one simultaneous release of one of those transitive dependencies (while the PR is open) might cause this workflow to fail.

I also think this might fail if the PR branch is behind `main`, afaik `github.base_ref` points to the latest commit on `main`.